### PR TITLE
[yang][202111] Fixing Ethertype field regex in acl rule yang to accept decimal values

### PR DESCRIPTION
**Pull request by dgsudharsan**
Cherry-pick https://github.com/Azure/sonic-buildimage/pull/10108

#### Why I did it
Fixing issue https://github.com/Azure/sonic-buildimage/issues/9991
The ACL RULE table field ETHER_TYPE can accept both hex as well as decimal values. However yang model didn't allow decimal values. Fixed it to allow decimal values (same pattern as in hex (1536-65535)

#### How I did it
Updated yang model to handle decimal values

#### How to verify it
Added UT to verify it.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

File: src/sonic-yang-models/yang-templates/sonic-acl.yang.j2 [hunk 1]
```diff
@@ -152,7 +152,7 @@ module sonic-acl {
 
 				leaf ETHER_TYPE {
 					type string {
-					    pattern "0x0[6-9a-fA-F][0-9a-fA-F]{2}|0x[1-9a-fA-F][0-9a-fA-F]{3}";
+                                            pattern "0x0[6-9a-fA-F][0-9a-fA-F]{2}|0x[1-9a-fA-F][0-9a-fA-F]{3}|153[6-9]|15[4-9][0-9]|1[6-9][0-9][0-9]|[2-9][0-9]{3}|[1-5][0-9]{4}|6553[0-5]|655[0-2][0-9]|65[0-4][0-9]{2}|6[0-4][0-9]{3}";
```
Comment (qiluo-msft) [inline src/sonic-yang-models/yang-templates/sonic-acl.yang.j2:155]: >                                             [](http://example.com/codeflow?start=0&length=44)

Mixing tab and spaces in the same file. Could you make them consistent?

Comment (dgsudharsan) [reply src/sonic-yang-models/yang-templates/sonic-acl.yang.j2]: The entire file is not properly intended. I can perform indentation but will show as multiple changes. Is that fine?

Comment (dgsudharsan) [reply src/sonic-yang-models/yang-templates/sonic-acl.yang.j2]: Aligned it as before

**Comment by qiluo-msft** [comment]
Possible reason for build failure: missing https://github.com/Azure/sonic-buildimage/pull/10043

**Review by ganglyu — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Comment by dgsudharsan** [comment]
@qiluo-msft Even after the change the build is failing in arm compilation and kvmtest. Can you please help?

**Comment by dgsudharsan** [comment]
@qiluo-msft @judyjoseph @ganglyu Can we get this merged?

**Pull request merged**